### PR TITLE
DC-631 Next Core package release

### PR DESCRIPTION
**Pull request by eltesoro-koji**
* I created a separate Jira for the Custom VCC package since that's in a separate repo and is not as rushed. See https://withkoji.atlassian.net/browse/DC-641.
* Also updated non-TypeDoc comments (i.e., those starting with '//') and error messages for consistency. Let me know if you want me to undo those.
* Because of lack of clarity with "configuration" vs "customization", let me know if you think I should use one instead of the other at specific places.
* Also tweaked some of the wording, including the verbs related to "focus".
* I'll probably need to do another pass (separate Jira) after we finalize the terminology.

### CODE REVIEW

File: README.md
Comment (rasienko) [inline README.md]:
```suggestion
Initialize the package with your configuration data, and use `KojiBackend.middleware` to scope operations per creator app.
```
Customization is a bit unclear here, but I'm not sure if my suggestion clarifies enough. The idea is that the middleware allows each creator's app to use its own data.

File: README.md
Comment (rasienko) [inline README.md]:
```suggestion
// Use Koji's middleware to handle scoping across customized versions of the app
```
Same as the previous comment. "Customizations" feels right when talking about the options provided to the creator. We need a way to talk about the customized app for each creator.

File: README.md
Comment (rasienko) [inline README.md]:
```suggestion
* [Migrating to the Koji core package](https://developer.withkoji.com/docs/dev-guides/migrate-koji-core)
```

Looks like the Gatsby site handles the link redirect, but let's update it to the latest since we're here.

File: src/backend/base/index.ts
Comment (rasienko) [inline src/backend/base/index.ts]:
```suggestion
  /** Express response object. Used in conjunction with middleware to scope environment variables for custom versions of the original Koji app. For the original definition see [[https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express/index.d.ts#L127 | @types/express]]. */
```

File: src/backend/identity/index.ts
Comment (rasienko) [inline src/backend/identity/index.ts]:
```suggestion
 * Defines a notification to send to a user's Koji account.
```
The notifications are sent to the user's account on the Koji platform, so reverting to this clarification.

File: src/backend/identity/index.ts
Comment (rasienko) [inline src/backend/identity/index.ts]:
```suggestion
 * Use [[pushNotificationToOwner]] to send notifications to the creator of the app.
```
Technically, they don't need to post it to their profile, so we can simplify this language a bit here.

File: src/backend/identity/index.ts
Comment (rasienko) [inline src/backend/identity/index.ts:83]: Please revert this change. The notification goes to the Koji account of the user.

File: src/backend/identity/index.ts
Comment (rasienko) [inline src/backend/identity/index.ts]: Please revert. See previous comments.

File: src/backend/middleware/index.ts
Comment (rasienko) [inline src/backend/middleware/index.ts]:
```suggestion
    // Apply version-specific overrides (equivalent of window.KOJI_OVERRIDES.overrides on the client)
```

Again, "customization" isn't quite right, but not sure my fix is either.

File: src/frontend/analytics/index.ts
Comment (rasienko) [inline src/frontend/analytics/index.ts]: I believe this navigation path has changed (since we got rid of the Koji button).

File: src/frontend/index.ts
Comment (rasienko) [inline src/frontend/index.ts]: Again, customization doesn't feel like the right word here. Update to whatever we decide.

File: src/frontend/playerState/index.ts
Comment (rasienko) [inline src/frontend/playerState/index.ts]: Both values are sent when someone is "customizing" that app. `instant` means in the player and `full` means in the code editor. The code editor way is really not a thing anymore, so we might ask Sean if he wants to remove this and related properties from a future version of the package/platform.  For now, maybe something like...
```suggestion
 * Type of editor the creator is using, either `instant` for the Koji player or `full` for the Koji code editor.
```

File: src/frontend/playerState/index.ts
Comment (rasienko) [inline src/frontend/playerState/index.ts]: It describes the editor that the creator is using (this interface provides information to the app).

File: src/frontend/playerState/index.ts
Comment (rasienko) [inline src/frontend/playerState/index.ts]: Match changes for EditorType

File: src/frontend/remix/index.ts
Comment (rasienko) [inline src/frontend/remix/index.ts]: Remove reference to profile here. It's just the creator of the app (whether it's currently on their profile or not).

File: src/frontend/ui/capture/index.ts
Comment (rasienko) [inline src/frontend/ui/capture/index.ts]:
```suggestion
  /** If the resource is a Koji app, the app's name, if available. */
```

File: src/backend/iap/index.ts
Comment (rasienko) [inline src/backend/iap/index.ts:116]: For the `owner`, we could update the description to say "Object that represents the app creator ("seller")."

File: src/frontend/analytics/index.ts
Comment (rasienko) [inline src/frontend/analytics/index.ts]:
```suggestion
 * You will see data for the custom events along with a standard set of built-in metrics, which are calculated from the app's access logs.
```

File: src/frontend/playerState/index.ts
Comment (rasienko) [inline src/frontend/playerState/index.ts]:
```suggestion
   * @param isRemixing Indicates whether the app is in customization mode.
```

**Review by rasienko — CHANGES_REQUESTED** [review]
Great stuff! It seems like you were able to access everything and follow the TypeDoc “processes”.  It was a good idea to update all the comments. Even if the // comments are less of a priority, it is still nice to have them match. Thanks!

A few minor comments here and there, but it seems like you caught almost all of the “template” and “Koji” instances. That was not an easy task!

FYI, I already updated the repo description to match your changes to the readme file.

Things to left to do: 
- Resolve comments & regenerate docs JSON file
- package.json - make sure that description matches the updated language in the readme and update the version to 1.7.4 (coordinate with Sean before he merges all of the open PRs).
- Open a PR against the docs repo to make matching changes to the core overview static page (per changes to the readme file), the frontend static page (per changes to src/frontend/index.ts and to the class descriptions), and the backend static page (per changes to the class descriptions). You’ll want to merge this PR after the core package is released (so that Gatsby does a fresh build of the package contents).

Let me know if you have questions. Great job!

File: src/frontend/remix/index.ts
Comment (eltesoro-koji) [reply src/frontend/remix/index.ts]: "creator of the app" makes it easily confused with the developer of the app. What I understood is that we talk about three personas: the developer of the app, the social media creator who adds the app to their profile, and the user who interacts with the app. So "creator of the app" is vague.

File: README.md
Comment (eltesoro-koji) [reply README.md]: Fixed.

File: README.md
Comment (eltesoro-koji) [reply README.md]: Fixed.

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts]: Reverted.

File: src/frontend/remix/index.ts
Comment (eltesoro-koji) [reply src/frontend/remix/index.ts]: I thought we chose "creator" because they are social media creators, not because they are Koji app creators, which is false. The developers created the app; the "creators" only customized the app. Maybe we need a different word besides "creator".

File: README.md
Comment (eltesoro-koji) [reply README.md]: BEFORE: ... to scope operations per customization.
AFTER: ... to scope operations for each customized version of the app.

File: src/backend/base/index.ts
Comment (eltesoro-koji) [reply src/backend/base/index.ts]: Fixed.

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts]: I'm not sure I understand this. Twitter doesn't say their notifications go to the Twitter account of their user; it's a given. Where do the notifications go if not the account of the user?

File: src/backend/middleware/index.ts
Comment (eltesoro-koji) [reply src/backend/middleware/index.ts]: Went with "version-specific". It's an internal comment anyway, so not high priority.

File: src/frontend/index.ts
Comment (eltesoro-koji) [reply src/frontend/index.ts]: Replaced with "customized versions".

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: How about we just say, "Must be set to `instant`." until Sean removes this option?

Using the code editor means they have to fork the project, so the "creator" becomes a "developer" at that point, and it's an entirely different scenario then.

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: Same here. We should just say that it returns `instant` and `edit`, because `full` and `new` should not be available anymore.

File: src/frontend/ui/capture/index.ts
Comment (eltesoro-koji) [reply src/frontend/ui/capture/index.ts]: Fixed.

File: src/frontend/analytics/index.ts
Comment (eltesoro-koji) [reply src/frontend/analytics/index.ts]: Fixed.

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: Fixed.

File: src/backend/iap/index.ts
Comment (eltesoro-koji) [reply src/backend/iap/index.ts:116]: Fixed.

Also replaced the description of *ownerUserId* with "Koji username of the seller."

File: src/frontend/analytics/index.ts
Comment (eltesoro-koji) [reply src/frontend/analytics/index.ts]: Fixed.

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts]: Confusion between "developer" and "creator" here.
My understanding is that the "developer" forks a repository to modify the code to create a new app. On the other hand, a creator only customizes (or creates custom versions) of the app by changing its settings or metadata and/or by replacing the app's assets. The creator does not touch the code directly.

File: src/frontend/remix/index.ts
Comment (sthielen) [reply src/frontend/remix/index.ts]: Maybe we can eliminate confusion in these package comments by only referring to a user's role in relationship to the app, rather than their role within the broader Koji ecosystem? 

At runtime, the app itself doesn't care whether the user is a developer-creator/social-media-creator/etc. (as each instance of the app is fully isolated from other instances) - would avoiding `creator` entirely standardizing on something like `owner` or `admin` make things more clear within the package docs?

File: src/frontend/playerState/index.ts
Comment (rasienko) [reply src/frontend/playerState/index.ts]: This value is not *set*. It is returned to a callback method so the developer could use it to tailor the experience based on the editor type. So, best to describe the possible values until/unless we remove them.

File: src/frontend/playerState/index.ts
Comment (rasienko) [reply src/frontend/playerState/index.ts]: See previous

File: src/frontend/remix/index.ts
Comment (rasienko) [reply src/frontend/remix/index.ts]: I like this perspective of discussing these values in relationship to the app. I do think we need to define what "owner" or "admin" mean in relation to how we discuss terms elsewhere in the docs. I believe "creator" is the correct term here because that is who can decrypt values on the frontend (whether they published this as a non-developer or a developer). Then, we can use "owner" as the technical term for the reference docs.

Maybe something like "Only the Koji creator ("owner") can ..."

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: Replaced.

File: src/frontend/remix/index.ts
Comment (eltesoro-koji) [reply src/frontend/remix/index.ts]: I like "owner", which can be either the developer or the social-media creator.

We already use "admin" as a special case of "user", i.e., someone who can access the admin context if there is one in the app.

I'll add "owner" in the glossary as someone who owns a specific version of the app. The developer owns the original version of the app; the creator who customized the app owns that specific customized version.

Comment (sthielen) [reply src/frontend/remix/index.ts]: Even though it might not be the case long term (e.g., if there end up being "teams" or "organizations"), today, the admin and owner user will always be the same. I think that's going to be the case for the foreseeable future, so I wouldn't be opposed to using a single term for internal documentation.

From the app's perspective, it only ever has one "creator" - the user who owns that App ID in Koji's Database. That "creator" might be the person who wrote the original code, or it might be the person who customized it; but by design there is no case where an app needs to understand that it might have other clones, or a parent, that are owned by different users.

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts]: Replaced with "owner".

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: Fixed.

File: src/frontend/playerState/index.ts
Comment (eltesoro-koji) [reply src/frontend/playerState/index.ts]: Fixed.

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts:83]: Per Slack conversation with Rachel, not reverting. The distinction she intended is related to the source of the notification (platform vs app), not the target (user vs user account).

File: src/backend/identity/index.ts
Comment (eltesoro-koji) [reply src/backend/identity/index.ts]: See previous comment.

File: src/frontend/remix/index.ts
Comment (eltesoro-koji) [reply src/frontend/remix/index.ts]: I think it would create more confusion if we use the same term ("creator") to mean two different things within the same documentation set. These TypeDoc docs end up in the same website as the AsciiiDoc docs where we use "creator" to mean the people who customize the app and "developer" to mean the coder of the original app.

Maybe we need a different word to mean "customizer". One that's more elegant than "customizer".

**Comment by rasienko** [comment]
@eltesoro-koji Looks great! All text changes are approved. A few things to do before the PR is approved and ready to merge:

- Regenerate the docs JSON file
- In package.json - update the description to match the updated language in the readme and update the version to 1.7.4

When this PR is done, we'll open a PR against the docs for the remaining work (basically, updating the static content and then merging that after the package release is finalized to trigger a new build of the Gatsby site).

**Review by rasienko — APPROVED** [review]
Looks great, thanks!

**Review by sthielen — APPROVED** [review]

**Pull request merged**
